web/js/holidayManagement.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2021 Igalia, S.L. <info@igalia.com>
+ *
+ * This file is part of PhpReport.
+ *
+ * PhpReport is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PhpReport is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PhpReport.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var app = new Vue({
+    el: '#holidaysApp',
+    data() {
+        return {
+            days: [],
+            range: {},
+            ranges: [],
+            total: null,
+
+            // Clean selected range styles to avoid confusion when
+            // removing dates
+            selectAttribute: {
+                highlight: {
+                    start: {
+                        style: {
+                            backgroundColor: 'transparent',
+                        },
+                    },
+                    base: {
+                        style: {
+                            backgroundColor: 'transparent',
+                        }
+                    },
+                    end: {
+                        style: {
+                            backgroundColor: 'transparent',
+                        }
+                    },
+
+                },
+            },
+        };
+    },
+    created() {
+        const fetchData = async () => {
+            const currentYear = new Date().getFullYear();
+            const url = `/web/services/getHolidays.php?initi=${currentYear}-01-31&end=${currentYear}-12-31`;
```

Comment:
I have my local dev environment at http://localhost/phpreport, and this component is not working because it's calling http://localhost/web/services/getHolidays.php

I had to modify this url to start with: `services/getHolidays.php...`

Reply:
Oh I see.. just changed to `services/...` so we don't depend on the app setup